connect-examples/oauth/php/callback.php

```diff
@@ -7,59 +8,54 @@
 #    is already signed in to Square, and if the merchant has already authorized your application,
 #    the OAuth flow automatically proceeds to the next step without presenting the Permissions form.
 # 3. Square sends a request to your application's Redirect URL
-#    (which should be set to http://localhost:8080/callback.php on your application dashboard)
+#    (which should be set to http://localhost:8000/callback.php on your application dashboard)
 # 4. The server extracts the authorization code provided in Square's request and passes it
 #    along to the Obtain Token endpoint.
 # 5. The Obtain Token endpoint returns an access token your application can use in subsequent requests
 #    to the Connect API.
 #
-# This sample requires the Unirest PHP library. Follow the instructions in README.md
-# to install it.
 
 require 'vendor/autoload.php';
 
 # Your application's ID and secret, available from your application dashboard
 $applicationId = 'REPLACE_ME';
 $applicationSecret = 'REPLACE_ME';
 
-$connectHost = 'https://connect.squareup.com';
-
-# Headers to provide to OAuth API endpoints
-$oauthRequestHeaders = array (
-  'Authorization' => 'Client ' . $applicationSecret,
-  'Accept' => 'application/json',
-  'Content-Type' => 'application/json'
-);
+$oauth_api = new SquareConnect\Api\OAuthApi();
 
 # Serves requests from Square to your application's redirect URL
 # Note that you need to set your application's Redirect URL to
 # http://localhost:8000/callback.php from your application dashboard
 function callback() {
-  global $connectHost, $oauthRequestHeaders, $applicationId, $applicationSecret;
+  global $oauth_api, $applicationId, $applicationSecret;
 
   # Extract the returned authorization code from the URL
   $authorizationCode = $_GET['code'];
   if ($authorizationCode) {
 
-    # Provide the code in a request to the Obtain Token endpoint
-    $oauthRequestBody = array(
-      'client_id' => $applicationId,
-      'client_secret' => $applicationSecret,
-      'code' => $authorizationCode
-    );
-    $response = Unirest\Request::post($connectHost . '/oauth2/token', $oauthRequestHeaders, json_encode($oauthRequestBody));
+    # Create Obtain Token requests
+    $body = new \SquareConnect\Model\ObtainTokenRequest();
+    $body->setClientId($applicationId);
+    $body->setClientSecret($applicationSecret);
+    $body->setCode($authorizationCode);
+
+    try {
+      $response = $oauth_api->obtainToken($body);
 
-    # Extract the returned access token from the response body
-    if (property_exists($response->body, 'access_token')) {
+      # Extract the returned access token from the response body
+      if ($response->getAccessToken() != null) {
```

Comment:
Can you refine your code a little bit to only call "getAccessToken()" once?

Reply:
Updated